examples(oracle): add tutorial DAG using OracleHook to fetch/sample rows

mangal-vairalkar opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---

### Summary
This PR adds a new example DAG demonstrating how to use the `OracleHook` 
to connect to an Oracle database and fetch/sample rows. 
It provides a simple ingestion workflow that can be adapted by users 
for Oracle-to-Airflow pipelines.

### Changes
- Added `example_oracle_ingest.py` under `airflow/providers/oracle/example_dags/`.
- Shows how to:
  - Establish a connection with Oracle using `OracleHook`.
  - Run a sample SQL query.
  - Push results to XCom for downstream tasks.

### Why
Currently, Oracle provider users have limited DAG-level examples.  
This example helps new users quickly understand how to use OracleHook 
in a production DAG setup.

### Tests
- Verified DAG loads successfully with `airflow dags list`.
- DAG can connect to an Oracle instance (requires valid connection ID).
- Syntax and pre-commit checks pass.

### Related
N/A (new example, no issue reference)


**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

mangal-vairalkar committed:
examples(oracle): add tutorial DAG using OracleHook

prdai commented on providers/oracle/example_dags/example_oracle_ingest.py:
I think it would be nice if the functions in this example DAG had type hints.

For example:

```python
def fetch_rows() -> list[tuple[int, str]]:
    ...

def print_rows(ti: TaskInstance | None = None) -> None:
    ...
```

mangal-vairalkar committed:
refactor(oracle): move example DAG to providers/src layout

jscheffl commented on providers/src/airflow/providers/oracle/example_dags/example_oracle_ingest.py:
You call this "example oracle **ingest**" but actually you make a fetch in the example? Do you mind renaming the file accordingly? (and the Dag name as well)

jscheffl reviewed (commented):
In general looks good but am wondering about the naming a bit.

mangal-vairalkar committed:
examples(oracle): rename DAG and file to example_oracle_fetch

mangal-vairalkar committed:
Update example_oracle_fetch.py

This PR renames the Oracle example DAG from example_oracle_ingest to example_oracle_fetch in both the file name and DAG name.

🔎 Context

During review it was noted that the example does not perform an ingest but rather a fetch operation. To avoid confusion and align with the functionality, the naming has been updated.

✅ Changes Made

Renamed file:
example_oracle_ingest.py → example_oracle_fetch.py

Updated DAG ID and related references accordingly.

Ensured consistency in code comments and references.

🧪 Validation

Pre-commit checks and CI validators have passed successfully.

Verified DAG loads and runs with the updated name.

🎯 Motivation

Improves clarity and accuracy of example DAGs.

Keeps naming consistent with expected behavior (fetch instead of ingest).

jscheffl reviewed (approved):
Thanks! Yeah looks clean now and naming is also adjusted! Thanks for the contribution!

Let's have CI runnign and then I assume it is good to merge.
